specs/opcode/F3RETURN_FDREVERT.md

```diff
@@ -68,13 +78,18 @@ using a lookup to the copy circuit with the following parameters:
   rwc_inc: 2 * min(return_length, call_context[ReturnDataLength, callee_id]),
 }
 ```
-If `min(return_length, call_context[ReturnDataLength, callee_id])` is zero, it does not perform the copy circuit lookup.
+If `min(return_length, call_context[ReturnDataLength, callee_id])` is zero, it
+does not perform the copy circuit lookup.
+
+**E** - Reverting the state changes is handled by the `ReversionInfo` used when
```

Comment:
When reverting the state, we must constraint that the `rw_counter` change to the next state considers the `instruction.curr.reversible_write_counter` (plus the the count of reversible operations that happened in this step, if any).
I think you skipped that in the python part.

See this for an example of a step that reverts: https://github.com/privacy-scaling-explorations/zkevm-specs/blob/33e78c0d5f3baae05d6a51cedc6df387efa48532/src/zkevm_specs/evm/execution/oog_constant.py#L33